In [105]:
import pandas as pd
acc = pd.read_csv('202110_CPGF.csv',sep = ';',encoding='ISO-8859-1', decimal = ",")
acc.head(1)

CÓDIGO ÓRGÃO SUPERIOR  ... VALOR TRANSAÇÃO
0                  63000  ...           670.0

[1 rows x 15 columns]

##Item K

In [106]:

acc['VALOR TRANSAÇÃO'].sum()

5619007.95

A soma total das transações é de R$5.619.007,95.

##Item L

In [107]:
acc[acc['NOME PORTADOR'] == 'Sigiloso']['VALOR TRANSAÇÃO'].sum()

3108731.15

A soma das transações de cunho sigiloso é de R$3.108.731,15

##Item M

In [108]:
acc[acc['NOME PORTADOR'] == 'Sigiloso'].groupby('NOME ÓRGÃO').sum()['VALOR TRANSAÇÃO'].sort_values(ascending = False)

NOME ÓRGÃO
Presidência da República                                1699751.04
Departamento de Polícia Federal                         1207131.92
Gabinete da Vice-Presidência da República                140238.19
Ministério da Economia - Unidades com vínculo direto      24000.00
Departamento de Polícia Rodoviária Federal                19500.00
Fundo Constitucional do Distrito Federal                  18110.00
Name: VALOR TRANSAÇÃO, dtype: float64

O órgão que mais realizou movimentações sigilosas foi a Presidência da República, com o total de R$1.699.751,04.

##Item N

Lista dos portadores que mais realizaram saques, em ordem decrescente:

In [109]:
acc[acc['TRANSAÇÃO'] == 'SAQUE CASH/ATM BB'].groupby('NOME PORTADOR').count()['CPF PORTADOR'].sort_values(ascending = False)

NOME PORTADOR
RAFAEL FERREIRA COSTA                    25
RONISONCICLEY CONDE DA ROCHA             12
ARTEMIO DE BRITO MACHADO                  9
FRANCISCO CHAGAS DOS SANTOS FILHO         8
MARIAN HELEN DA SILVA GOMES RODRIGUES     8
                                         ..
HIRAN HERNANE HURPIA                      1
RAFAEL MARCONI DOS SANTOS                 1
HILDEBRANDO FERREIRA FILHO                1
ADMAR HELINTON DORNFELD                   1
SEM INFORMACAO                            0
Name: CPF PORTADOR, Length: 351, dtype: int64

Portanto, o portador que mais realizou saques foi Rafael Ferreira Costa, com 25 saques no mês de outubro de 2021.

O órgão ao qual esse portador pertence é encontrado abaixo:

In [110]:
acc[acc['NOME PORTADOR'] == 'RAFAEL FERREIRA COSTA']['NOME ÓRGÃO'].iloc[0]

'Instituto Chico Mendes de Conservação da Biodiversidade'

O órgão ao qual o portador Rafael Ferreira Costa pertence é o Instituto Chico Mendes de Conservação da Biodiversidade.

##Item O

In [111]:
acc[acc['TRANSAÇÃO'] == 'COMPRA A/V - R$ - APRES'].groupby('NOME FAVORECIDO').count()['CPF PORTADOR'].sort_values(ascending = False)

NOME FAVORECIDO
MERCADOPAGO.COM REPRESENTACOES LTDA.                 121
SEM INFORMACAO                                        97
A.M. COM. DE DERIVADOS DE PETROLEO LTDA               32
PAGAR.ME PAGAMENTOS S.A.                              29
KALUNGA COMERCIO E INDUSTRIA GRAFICA LTDA             29
                                                    ... 
CASA DO CONSTRUTOR E MATERIAIS DE CONSTRUCAO LTDA      0
HOT DIGITAL COMERCIO E SERVICO DE IMPRESSAO LTDA       0
VANDERSON VERVLOET BESSA COMERCIO DE GAS               0
ADILSON DE OLIVEIRA MEDEIROS                           0
POSTO 7 ARARAS LTDA                                    0
Name: CPF PORTADOR, Length: 3523, dtype: int64

O favorecido que mais recebeu compras utilizando o CGPF foi o Mercadopago.com Representações Ltda, com 121 compras no total.